src/bench/sink_bench/main.rs

```diff
@@ -0,0 +1,463 @@
+                is_visible: true,
+            })
+            .collect()
+    }
+
+    pub fn get_sink_schema(&self) -> Vec<ColumnDesc> {
+        self.columns
+            .iter()
+            .map(|column| {
+                ColumnDesc::named(column.name.clone(), ColumnId::new(1), column.r#type.clone())
+            })
+            .collect()
+    }
+}
+#[derive(Debug, Deserialize)]
+struct ColumnDescFromYml {
+    name: String,
+    #[serde(deserialize_with = "deserialize_datatype")]
+    r#type: DataType,
+}
+
+fn deserialize_datatype<'de, D>(deserializer: D) -> Result<DataType, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+    DataType::from_str(s).map_err(serde::de::Error::custom)
+}
+
+fn read_table_schema_from_yml(path: &str) -> TableSchemaFromYml {
+    let data = std::fs::read_to_string(path).unwrap();
+    let table: TableSchemaFromYml = serde_yaml::from_str(&data).unwrap();
+    table
+}
+
+fn read_sink_option_from_yml(path: &str) -> HashMap<String, HashMap<String, String>> {
+    let data = std::fs::read_to_string(path).unwrap();
+    let sink_option: HashMap<String, HashMap<String, String>> =
+        serde_yaml::from_str(&data).unwrap();
+    sink_option
+}
+
+#[derive(Parser, Debug)]
+pub struct Config {
+    #[clap(long, default_value = "./sink_bench/schema.yml")]
+    schema_path: String,
+
+    #[clap(short, long, default_value = "./sink_bench/sink_option.yml")]
+    option_path: String,
+
+    #[clap(short, long, default_value = BENCH_TEST)]
+    sink: String,
+
+    #[clap(short, long)]
+    rows_per_second: u64,
+
+    #[clap(long, default_value = "10")]
+    split_num: String,
+}
+
+fn mock_from_legacy_type(
+    connector: &str,
+    r#type: &str,
+) -> Result<Option<SinkFormatDesc>, SinkError> {
+    use risingwave_connector::sink::redis::RedisSink;
+    use risingwave_connector::sink::Sink as _;
+    if connector.eq(RedisSink::SINK_NAME) {
+        let format = match r#type {
+            SINK_TYPE_APPEND_ONLY => SinkFormat::AppendOnly,
+            SINK_TYPE_UPSERT => SinkFormat::Upsert,
+            _ => {
+                return Err(SinkError::Config(risingwave_common::array::error::anyhow!(
+                    "sink type unsupported: {}",
+                    r#type
+                )))
+            }
+        };
+        Ok(Some(SinkFormatDesc {
+            format,
+            encode: SinkEncode::Json,
+            options: Default::default(),
+        }))
+    } else {
+        SinkFormatDesc::from_legacy_type(connector, r#type)
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    let cfg = Config::parse();
+    let table_schema = read_table_schema_from_yml(&cfg.schema_path);
+    let mock_datagen_source = MockDatagenSource::new(
+        cfg.rows_per_second,
+        table_schema.get_source_schema(),
+        cfg.split_num,
+    )
+    .await;
+    let throughput_metric = std::sync::Arc::new(RwLock::new(ThroughputMetric::new()));
```

Comment:
I am worried that acquiring the lock every time we collect the metric may affect the performance.

I think we can own the `ThroughputMetrics` in the tokio task that run the sink benchmark. The tokio task can hold a oneshot rx to get notified on benchmark finish with a `select`, and then it can return the metrics as the task output. The  `main` can get the metrics via awaiting on the tokio task join handle.